pkg/api/microservices/microservices_test.go

```diff
@@ -0,0 +1,159 @@
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// TestReturnAllMicroservices tests that all Microservices are returned
+// GIVEN that Microservices are configured
+// WHEN I call GetAllMicroservices
+// THEN all the Microservices should be returned
+// AND that the Microservices returned should be the correct amount and have valid values
+func TestReturnAllMicroservices(t *testing.T) {
+	var modelBindingPairs = map[string]*types.ModelBindingPair{
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	// GIVEN empty model binding pairs, managed clusters and fake kubernetes clients.
+	var clients kubernetes.Interface = k8sfake.NewSimpleClientset()
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+
+	type args struct {
+		w http.ResponseWriter
+		r *http.Request
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{
+			name:    "verifyReturnAllMicroservices",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var myMicroservices []Microservice
+			Init(tt.listers)
+			resp := httptest.NewRecorder()
+			ReturnAllMicroservices(resp, httptest.NewRequest("GET", "/microservices", nil))
+			assert.Equal(t, http.StatusOK, resp.Code, "expect the http return code to be http.StatusOk")
+			json.NewDecoder(resp.Body).Decode(&myMicroservices)
+			assert.Len(t, myMicroservices, 2, "expect returned MicroServices Array to have 2 entries")
+			assert.Equal(t, "HelidonApp1", myMicroservices[0].Name, "expect Helidon App Name to be HelidonApp1")
+			assert.Equal(t, "cluster1", myMicroservices[0].Cluster, "expect Helidon App Cluster Name to be cluster1")
+			assert.Equal(t, "defaultOnMC1", myMicroservices[0].Namespace, "expect Helidon App Namespace to be defaultOnMC1")
+			assert.Equal(t, "HelidonApp2", myMicroservices[1].Name, "expect Helidon App Name to be HelidonApp2")
+			assert.Equal(t, "cluster2", myMicroservices[1].Cluster, "expect Helidon App Cluster Name to be cluster2")
+			assert.Equal(t, "defaultOnMC2", myMicroservices[1].Namespace, "expect Helidon App Namespace to be defaultOnMC2")
+		})
+	}
+}
+
+// TestReturnSingleMicroservice tests that the Microservice with ID=HelidonApp1 is returned
+// GIVEN that Microservice with ID=HelidonApp1 exists
+// WHEN I call ReturnSingleMicroservice
+// THEN the Microservice with ID=HelidonApp1 should be returned
+// AND that the Microservice returned should have the correct ID and have valid values
+func TestReturnSingleMicroservice(t *testing.T) {
+	var modelBindingPairs = map[string]*types.ModelBindingPair{
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	// GIVEN empty model binding pairs, managed clusters and fake kubernetes clients.
+	var clients kubernetes.Interface = k8sfake.NewSimpleClientset()
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+
+	type args struct {
+		w http.ResponseWriter
+		r *http.Request
+	}
+	tests := []struct {
```

Comment:
So I thought the point of this set up with the tests struct was to run multiple cases in the same test. This tests list could easily have 2 different cases - one "verifyReturnSIngleMicroservice" case which expects a certain microservice object, and another which expects one not to be found

Reply:
If you prefer it, ur wish is my command

Reply:
I'm going to be careful what I wish for ;) 
I don't want to hold up this PR on what is essentially a matter of preference. But I am not too fond of using that elaborate test struct with args, if all we are going to do is test a single test case. Then it's a lot of set up for just one test case.